### PR TITLE
ARROW-5805: [Python] Dockerize (add to docker-compose) Python Travis CI job

### PR DESCRIPTION
**Pull request by rok**
This is to resolve [ARROW-5805](https://issues.apache.org/jira/browse/ARROW-5805).

### CODE REVIEW

**Review by wesm — COMMENTED** [review]
@fsaintjacques can you coordinate this patch with the one you just put up for the C++ build?

**Comment by fsaintjacques** [comment]
@wesm the `cpp-ci` image found in ARROW-5803 does not use conda, do you want me to convert this python image to avoid conda _or_ tackle [ARROW-5804](https://jira.apache.org/jira/browse/ARROW-5804) which is cpp with conda.

**Comment by wesm** [comment]
@fsaintjacques Yes I think that we should have a combined C++/Python docker-compose entry that uses dependencies from conda

**Comment by fsaintjacques** [comment]
My suggestion also imply de-confa-ying the other images (since the base `arrow:cpp` is relying on conda), is the community fine with this (@xhochy, @kszucs)?

File: ci/conda_env_python.yml [hunk 1]
```diff
@@ -15,13 +15,13 @@
...
 # under the License.
 
+coverage
 cython=0.29.7
 cloudpickle
 hypothesis
 numpy>=1.14
 pandas
 pytest
-pytest-faulthandler
```
[the remark this answers is not in the document]
Comment (xhochy) [reply ci/conda_env_python.yml]: `pytest=5.0` includes this now

[the remark this answers is not in the document]
Comment (wesm) [reply ci/conda_env_python.yml]: ah, thanks

**Comment by xhochy** [comment]
> My suggestion also imply de-confa-ying the other images (since the base `arrow:cpp` is relying on conda), is the community fine with this (@xhochy, @kszucs)?
> 
> Note that this patch was initially implemented replacing cpp by removing conda, I ended up using `cpp-ci` instead to minimize the breaking changes.

Actually we should try to keep `conda` in most images as this is the ecosystem we can control best and it is our recommendation for the setup for Python users, our currently most popular user base.

**Comment by wesm** [comment]
How would you all like to proceed with this patch?

**Comment by kszucs** [comment]
@wesm I'm doing a pretty heavy undertaking against the docker-compose.yml
I can put up a PR tomorrow, it'd nice if you could wait for it.

**Comment by wesm** [comment]
No problem. I was just scanning through the patch queue

**Comment by rok** [comment]
@kszucs I'd be happy to continue working on this, but I'm not clear on the direction: are we keeping conda here or not? Should we have a python image with conda and one without?

**Comment by wesm** [comment]
Closing as this is superseded by the GitHub Actions patch

**Pull request closed without merging**